paddle/fluid/distributed/fleet_executor/carrier.h

```diff
@@ -26,31 +27,40 @@
 namespace paddle {
 namespace distributed {
 
-class Interceptor;
 class TaskNode;
 class InterceptorMessageServiceImpl;
 
+// A singleton MessageBus
 class Carrier final {
  public:
-  Carrier() = delete;
+  static Carrier& Instance() {
+    static Carrier carrier;
+    return carrier;
+  }
 
-  explicit Carrier(
+  void Init(
       const std::unordered_map<int64_t, TaskNode*>& interceptor_id_to_node);
 
   ~Carrier() = default;
 
   // Enqueue a message to corresponding interceptor id
   bool EnqueueInterceptorMessage(const InterceptorMessage& interceptor_message);
 
+  // get interceptor based on the interceptor id
+  Interceptor* GetInterceptor(int64_t interceptor_id);
+
+  // set interceptor with interceptor id
+  Interceptor* SetInterceptor(int64_t interceptor_id,
```

Comment:
return `void`? If `GetInterceptor` is pubic for new, we can let users first call `SetInterceptor` to set, then call `GetInterceptor` to get the ptr?

Reply:
For convenient.